homeassistant/components/sensor/postnl.py

```diff
@@ -0,0 +1,128 @@
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_USERNAME, CONF_PASSWORD,
+                                 ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['postnl_api==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+CONF_UPDATE_INTERVAL = 'update_interval'
+DEFAULT_NAME = 'postnl'
+ICON = 'mdi:package-variant-closed'
+ATTRIBUTION = 'Information provided by PostNL'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_UPDATE_INTERVAL, default=timedelta(seconds=1800)):
+    vol.All(cv.time_period, cv.positive_timedelta),
+})
+
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the PostNL platform."""
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    name = config.get(CONF_NAME)
+    update_interval = config.get(CONF_UPDATE_INTERVAL)
+
+    from postnl_api import PostNL_API
+
+    try:
+        api = PostNL_API(username, password)
+
+    except Exception:
+        _LOGGER.exception("Can't connect to the PostNL webservice")
+        return False
+
+    add_devices([PostNLSensor(api, name, update_interval)], True)
+
+
+class PostNLSensor(Entity):
+    """PostNL Sensor."""
+
+    def __init__(self, api, name, interval):
+        """Initialize the sensor."""
+        self.friendly_name = name
+        self._name = name
+        self._attributes = None
+        self._state = None
+
+        self._api = api
+
+        self.update = Throttle(interval)(self._update)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+
+        if self._state == 1:
+            return 'package'
+
+        return 'packages'
+
+    def _update(self):
+        """Update device state."""
+
+        shipments = self._api.get_relevant_shipments()
+        status_counts = {}
+
+        def parse_date(date):
+            return datetime.strptime(date.group(1)
+                                     .replace(' ', '')[:-6], '%Y-%m-%dT%H:%M:%S').strftime('%d-%m-%Y')
+
+        def parse_time(date):
+            return datetime.strptime(date.group(1)
+                                     .replace(' ', '')[:-6], '%Y-%m-%dT%H:%M:%S').strftime('%H:%M')
+
+        for shipment in shipments:
+            status = shipment['status']['formatted']['short']
+            status = re.sub(r'{(?:Date|dateAbs):(.*?)}', parse_date, status)
```

Comment:
Please include these parsing instructions in the `postnl_api` package and just return a parsed data structure to Home Assistant.

Reply:
I will add is as a separate helper function in `postnl_api`, however I won't enable it by default since it tampers with the default output and it is also a small performance hit. However, I agree that this kind of parsing logic shouldn't be in the Home Assistant plugin itself.